### PR TITLE
[enhancement] Improved-performance-and-accessibility #215

### LINKED ISSUE #215: [gsoc21] Lighthouse reports for Performance and Accessibility
Running the OpenWISP site through Lighthouse in Chrome dev tools shows that Accessibility (79) and Performance [32 (!) and 69] scores can be improved.

**Report files (HTML)**
[openwisp_lighthouse.zip](https://github.com/openwisp/openwisp-utils/files/6827515/openwisp_lighthouse.zip)

### Some low hanging fruit:

**Performance**
- https://web.dev/uses-long-cache-ttl/?utm_source=lighthouse&utm_medium=devtools
- https://web.dev/uses-rel-preload/?utm_source=lighthouse&utm_medium=devtools

**Accessibility**

- https://web.dev/color-contrast/?utm_source=lighthouse&utm_medium=devtools

### PR DESCRIPTION
**Pull request by niteshsinha17**
Improved accessibility of the main menu and filters
accessibility

- [x] I have read the [contributing guidelines](http://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly).
- [x] I have manually tested the proposed changes.
- [ ] I have written new test cases to avoid regressions. (if necessary)
- [ ] I have updated the documentation. (e.g. README.rst)
- [ ] I have added [change!] to commit title to indicate a backward incompatible change. (if required)
- [ ] I have checked the links added / modified in the documentation.

closes #215

### CODE REVIEW

File: openwisp_utils/admin_theme/templates/admin/base_site.html [hunk 1]
```diff
@@ -18,32 +18,33 @@
 
 {% block menu-bar %}
   <div class="head">
-    <span class="hamburger"><span></span></span>
+    <span tabindex="0" role="button" aria-label="Maxamize menu"class="hamburger"><span></span></span>
```
Comment (nemesisdesign) [inline openwisp_utils/admin_theme/templates/admin/base_site.html:21]: This line has 2 issues:

- "Maxamize menu" should be flagged as translatable
- "class" does not have any preceding whitespace

File: openwisp_utils/admin_theme/templatetags/ow_tags.py [hunk 2]
```diff
@@ -15,3 +16,9 @@ def ow_create_filter(cl, spec, total_filters):
             'show_button': total_filters > 4,
         }
     )
+
+
+@register.filter
+@stringfilter
+def join_string(value):
+    return value.lower().replace(' ', '-')
```
Comment (nemesisdesign) [inline openwisp_utils/admin_theme/templatetags/ow_tags.py:24]: can you please add a docstring to this filter to explain how it is used and why we need it? It's not super clear to me.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Tested it and looks good!
I found a couple minor issues below.

@nickberry17 can you double check and run the reporting tool again please?

File: openwisp_utils/admin_theme/templatetags/ow_tags.py [hunk 3]
```diff
@@ -21,4 +21,7 @@ def ow_create_filter(cl, spec, total_filters):
 @register.filter
 @stringfilter
 def join_string(value):
+    '''
+    Can be used to join strings with "-" to make id or class
+    '''
```
Comment (nemesisdesign) [inline openwisp_utils/admin_theme/templatetags/ow_tags.py:26]: pelase use double quotes for docstrings, `""""` for consistency with the rest of the codebase

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

File: openwisp_utils/admin_theme/templatetags/ow_tags.py (hunk 3 above)
Comment (niteshsinha17) [reply openwisp_utils/admin_theme/templatetags/ow_tags.py:26]: I did this again. very bad

**Review by atb00ker — COMMENTED** [review]
~It's not working for me as in the GIF.
What keys are you pressing?
I am pressing tab and it doesn't focus on menu groups, just the first two menu items and leaves the menu.

Tested on both FF and Chrome! :-(

Is anyone able to reproduce the GIF?~

Staging environment is working, something is wrong with my local, but tabs are working!

**Review by nemesisdesign — APPROVED** [review]
Ok, good enough for me, let's merge! :+1: 

@atb00ker thanks for checking! I do not see any problem right now.

**Pull request merged**